subprojects/docs/src/snippets/tasks/incrementalBuild-incrementalBuildAdvanced/groovy/buildSrc/src/main/java/org/example/Instrument.java

```diff
@@ -17,9 +19,12 @@ public abstract class Instrument extends DefaultTask {
     @OutputDirectory
     public abstract DirectoryProperty getDestinationDir();
 
+    @Inject
+    protected abstract FileSystemOperations getFs();
+
     @TaskAction
     public void doIt() {
-        getProject().copy(spec -> spec.
+        getFs().copy(spec -> spec.
```

Comment:
👍 

This fix avoids accessing the `Project` object from the task instance, since it is not available when running with cached configuration. Instead, it uses the underlying service that provides similar functionality. See: https://docs.gradle.org/current/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution